build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,239 @@
+import { BaseCommand } from "../../base";
+import { Repository, getPreReleaseDependencies } from "../../lib";
+import { isReleaseGroup, ReleaseGroup, ReleasePackage } from "../../releaseGroups";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+const checkNoLocalChanges: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+};
+
+const checkDepsInstalled: CheckFunction = async (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+): Promise<CheckResult> => {
+	const packagesToCheck = isReleaseGroup(rgOrPkg)
+		? context.packagesInReleaseGroup(rgOrPkg)
+		: // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		  [context.fullPackageMap.get(rgOrPkg)!];
+
+	async function checkDeps(pkg: Package): Promise<boolean> {
+		const depsAreInstalled = await pkg.checkInstall(print);
+		// console.warn(`deps for ${pkg.name}: ${depsAreInstalled}`);
+		return depsAreInstalled;
+		// return pkg.checkInstall(false);
+	}
+
+	// eslint-disable-next-line unicorn/no-array-method-this-argument, @typescript-eslint/no-misused-promises
+	const installed = await async.every(packagesToCheck, checkDeps);
+	if (!installed) {
+		return {
+			result: false,
+			message: "Some dependencies aren't installed. Try installing dependencies manually.",
+			fixCommand: "pnpm install",
+		};
+	}
+
+	return { result: true };
+};
+
+const checkHasRemote: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const remote = await context.gitRepo.getRemote(context.originRemotePartialUrl);
+	if (remote === undefined) {
+		return {
+			result: false,
+			message: "No remote found that points to the microsoft/FluidFramework repo.",
+		};
+	}
+	return { result: true };
+};
+
+const checkBranchUpToDate: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const remote = await context.gitRepo.getRemote(context.originRemotePartialUrl);
+
+	let succeeded = false;
+	try {
+		succeeded = await context.gitRepo.isBranchUpToDate(
+			context.originalBranchName,
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			remote!,
+		);
+	} catch (error) {
+		return {
+			result: false,
+			message: (error as Error).message,
+		};
+	}
+
+	if (!succeeded) {
+		return {
+			result: false,
+			message: "Branch is out of date with the remote. Try pulling the latest from upstream.",
+			fixCommand: "git pull",
+		};
+	}
+	return { result: true };
+};
+
+const checkHasNoPrereleaseDependencies: CheckFunction = async (
```

Comment:
personally I find it easier to read top named level functions using the `function` keyword instead of the arrow function syntax. Might just be because I'm more used it, but even thats because we tend to do it that way more in this codebase.

Reply:
No particular reason; I'll make them `function`-decared.